lib/pure/collections/sequtils.nim

```diff
@@ -274,6 +313,13 @@ func minmax*[T](x: openArray[T]): (T, T) =
     if h < x[i]: h = x[i]
   result = (l, h)
 
+func minmax*[T](x: openArray[T], cmp: proc(a, b: T): int): (T, T) {.effectsOf: cmp.} =
+  ## The minimum and maximum values of `x`.
+  result = (x[0], x[0])
+  for i in 1..high(x):
+    if cmp(x[i], result[0]) < 0: result[0] = x[i]
+    if cmp(result[1], x[i]) < 0: result[1] = x[i]
```

Comment:
Maybe I'm missing something but I think you can improve this to only use a single call to `cmp` as it indicates the direction.